faithcomesbyhearing/dbp
# FCBHDBP-577 Check the method to translate a plan because the value of playlist item duration is not being filled

# Description
Updated the endpoint to translate a plan to include the playlist item duration value.

## Change Log

1. Updated the endpoint to always execute the method for calculating duration. Previously, this method was only executed when a plan was translated using a command.
2. Removed obsolete methods such as `createTranslatedPlaylistItems` and `getCodecMetadata`.
3. Removed unnecessary includes.
4. Refactored the `processVersesOnTransportStream` method to eliminate redundancy. This method was previously implemented in four different places and has now been centralized.
5. Further refactoring of `processVersesOnTransportStream` method involved replacing the slice method of the Collection class with the array_slice method. This change prevents alterations to the transportStream instance, which is crucial in cases where two playlists share the same `transportStream` instance.
6. Created a new service, `app/Services/Plans/PlaylistItemService.php`, to house all logic related to the `PlaylistItem` Model.

## Issue Link
https://fullstacklabs.atlassian.net/browse/FCBHDBP-577

## How Do I QA This
We should run the following postman tests and they have to pass. We need to be logged in to run the postman tests.
1. https://digitalbibleplatform.postman.co/workspace/DBP4-workspace~0a283f7f-06d9-467d-97f0-fef92d18bd3d/request/12519377-e3ffd74c-9afe-47be-a116-2af7b20d832a
2. https://digitalbibleplatform.postman.co/workspace/DBP4-workspace~0a283f7f-06d9-467d-97f0-fef92d18bd3d/request/12519377-72ff40da-00af-435a-9521-3beff5c3d804
